Minimal portability patches to build the vendored 2026-TR AnyPIA source with
modern clang/libc++. Applied by oracle/build/Makefile to the work tree after
CRLF->LF normalization; oracle/vendor/ is never modified.

1. linux/BitAnnual{.cpp,Inlines.h}: the class member is `theData` (BitAnnual.h)
   but the published linux files still reference `data`; also add the copy
   constructor that only the Windows build provided.
2. include/WorkerDataGeneral.h: libc++ bitset::operator[] const returns a
   proxy; use .test() so the static_cast to the Sex enum is legal.
3. linux/CommaBase.h: qualify std::ends.
4. source32/anypiab/{anypiabdoc.h,anypiab.cpp}: include piaout and declare
   TaxData/PiaOut members only under DETAILS (piaout is not vendored), and
   include only the driver header selected by ANYPIAB.

--- a/oactobjs32/include/WorkerDataGeneral.h	2026-08-22 13:17:56
+++ b/oactobjs32/include/WorkerDataGeneral.h	2026-08-22 13:17:46
@@ -524,7 +524,7 @@
   ///
   /// <returns>MALE (0) if male, FEMALE (1) if female.</returns>
   Sex::sex_type getSex() const
-  { return static_cast<Sex::sex_type>(theBits[SEX_BIT]); }
+  { return static_cast<Sex::sex_type>(theBits.test(SEX_BIT)); }
   /// <summary>Returns type of taxes for specified year.</summary>
   ///
   /// <remarks>Needs to be implemented by a child class.</remarks>
--- a/oactobjs32/linux/BitAnnual.cpp	2026-08-22 13:17:56
+++ b/oactobjs32/linux/BitAnnual.cpp	2026-08-22 13:17:46
@@ -10,7 +10,7 @@
 // GENERAL NOTE:
 // When adding functions to this class, or changing the implementation
 // of existing functions, be careful not to use square brackets ( [] )
-// to access 'data' unless you're really sure of what you're doing.
+// to access 'theData' unless you're really sure of what you're doing.
 // This is because there are different implementations of this class
 // depending on whether you're using NT or Linux. Instead, use the
 // setBit() and getBit() functions.
@@ -20,31 +20,39 @@
 /// <param name="newBaseYear">New base year.</param>
 /// <param name="newLastYear">New last year.</param>
 ///
-/// <remarks>On the linux system, data is a vector of BitPacked's -- so the
+/// <remarks>On the linux system, theData is a vector of BitPacked's -- so the
 /// constructor needs to know how many to create. On other systems (which
-/// conform to the standard and have a BitSet class built in), data is a
+/// conform to the standard and have a BitSet class built in), theData is a
 /// BitSet and doesn't need any parameters because the default constructor
 /// initializes all of the bits to zero.</remarks>
 BitAnnual::BitAnnual( int newBaseYear, int newLastYear ) :
-data((newLastYear - newBaseYear) / U_NUMBITS + 1),
+theData((newLastYear - newBaseYear) / U_NUMBITS + 1),
 baseYear(newBaseYear), lastYear(newLastYear)
 { }
 
 /// <summary>Returns true if any bit is set, false otherwise.</summary>
 ///
 /// <returns>True if any bit is set, false otherwise.</returns>
+
+/// <summary>Copy constructor (added for modern toolchains; the published
+/// bundle omits it on the linux side).</summary>
+BitAnnual::BitAnnual( const BitAnnual& bitAnnual ) :
+theData(bitAnnual.theData), baseYear(bitAnnual.baseYear),
+lastYear(bitAnnual.lastYear)
+{ }
+
 bool BitAnnual::any() const
 {
-   for (unsigned i = 0; i < data.size(); i++) {
-      if (data[i].any())
+   for (unsigned i = 0; i < theData.size(); i++) {
+      if (theData[i].any())
          return true;
    }
    return false;
 }
 
-/// <summary>Zeroes out all data values.</summary>
+/// <summary>Zeroes out all theData values.</summary>
 void BitAnnual::deleteContents()
 {
-   for (unsigned i = 0; i < data.size(); i++)
-      data[i].deleteContents();
+   for (unsigned i = 0; i < theData.size(); i++)
+      theData[i].deleteContents();
 }
--- a/oactobjs32/linux/BitAnnualInlines.h	2026-08-22 13:17:56
+++ b/oactobjs32/linux/BitAnnualInlines.h	2026-08-22 13:17:46
@@ -22,7 +22,7 @@
    }
 #endif
    int indexYear = year - baseYear;
-   return data[indexYear / U_NUMBITS].get(indexYear % U_NUMBITS);
+   return theData[indexYear / U_NUMBITS].get(indexYear % U_NUMBITS);
 }
 
 /// <summary>Sets one value of data.</summary>
@@ -41,7 +41,7 @@
    }
 #endif
    int indexYear = year - baseYear;
-   data[indexYear / U_NUMBITS].set(indexYear % U_NUMBITS, bit);
+   theData[indexYear / U_NUMBITS].set(indexYear % U_NUMBITS, bit);
 }
 
 #endif    // __BITANNUALINLINES_H
--- a/oactobjs32/linux/CommaBase.h	2026-08-22 13:17:56
+++ b/oactobjs32/linux/CommaBase.h	2026-08-22 13:17:46
@@ -12,7 +12,7 @@
 /// <summary>Use ostrstream for formatting output.</summary>
 #define OSTREAM std::ostrstream
 /// <summary>End of ostrstream designated by ends.</summary>
-#define ENDS start << ends
+#define ENDS start << std::ends
 /// <summary>Use freeze function to freeze the string.</summary>
 #define FREEZE(strm) strm.freeze(false)
 
--- a/source32/anypiab/anypiab.cpp	2026-08-22 13:17:56
+++ b/source32/anypiab/anypiab.cpp	2026-08-22 13:17:46
@@ -6,8 +6,11 @@
 // printed can be specified in function savecase, at the end of this file.
 
 #include <iostream>
+#if defined(ANYPIAB)
 #include "AnypiabDoc.h"
+#else
 #include "AnypiaTestDoc.h"
+#endif
 #include "Trace.h"
 #include "TextWriterTraceListener.h"
 
--- a/source32/anypiab/anypiabdoc.h	2026-08-22 13:17:56
+++ b/source32/anypiab/anypiabdoc.h	2026-08-22 13:17:46
@@ -4,7 +4,9 @@
 
 #include <iostream>
 #include "pia.h"
+#ifdef DETAILS
 #include "piaout/piaout.h"
+#endif
 #include "BaseYearNonFile.h"
 #include "AwbiDataNonFile.h"
 #include "AssumptionsNonFile.h"
@@ -40,8 +42,10 @@
       WorkerDataArray *widowDataArray;
       // Widow and other family members' pia data.
       PiaDataArray *widowPiaDataArray;
+#ifdef DETAILS
       TaxData *taxData;
       PiaOut *piaOut;
+#endif
    public:
       AnypiabDoc();
       ~AnypiabDoc();
